src/api/unified/symbol_manager.cpp

```diff
@@ -230,6 +237,71 @@ af_err AFSymbolManager::setBackend(af::Backend bknd) {
     }
 }
 
+af_err AFSymbolManager::addBackendLibrary(const char *lib_path) {
+    if ((newCustomHandleIndex + 1) > MAX_BKND_HANDLES) {
+        // No more space for an additional handle
+        UNIFIED_ERROR_LOAD_LIB();
+    }
+
+    string show_flag    = getEnvVar("AF_SHOW_LOAD_PATH");
+    bool show_load_path = show_flag == "1";
+
+    typedef af_err (*func)(int*);
+    LibHandle handle = nullptr;
+    if ((handle = loadLibrary(lib_path))) {
+        func count_func =
+            (func)getFunctionPointer(handle, "af_get_device_count");
+        if (count_func) {
+            int count = 0;
+            count_func(&count);
+            AF_TRACE("Device Count: {}.", count);
+            if (count == 0) {
+                // No available device for this backend
+                // Or loaded library is invalid (since function can't be found)
+                handle = nullptr;
+                UNIFIED_ERROR_LOAD_LIB();
+            }
+        }
+
+        if (show_load_path) { printf("Using %s\n", lib_path); }
+
+        bkndHandles[newCustomHandleIndex] = handle;
+        newCustomHandleIndex++;
+
+        return AF_SUCCESS;
+    }
+    else {
+        // loadLibrary failed, maybe because path is invalid or another reason
+        UNIFIED_ERROR_LOAD_LIB();
+    }
+}
+
+af_err AFSymbolManager::setBackendLibrary(int lib_idx) {
+    typedef af_err (*func)(af_backend*);
+    int actual_idx = lib_idx + NUM_BACKENDS;
+
+    if (actual_idx >= MAX_BKND_HANDLES) {
+        // lib_idx more than the capacity of bkndHandles
+        UNIFIED_ERROR_LOAD_LIB();
```

Comment:
I actually meant to follow up with you guys on this after replying on his comment, but thanks for clarifying a bit. I see though that `UNIFIED_ERROR_LOAD_LIB` doesn't take any arguments for displaying more info about the error, so even though I wanted to do this (even before Umar's comment), I really can't. Is there already an existing way that we could use to display a better error message? Otherwise I could start thinking of ways to do so (printf to stderr, or add a message through AF_TRACE, or modify `UNIFIED_ERROR_LOAD_LIB`, or add a new function, etc).

Reply:
`AF_TRACE` is fine I think.